### PR TITLE
Build/distributable

### PR DESCRIPTION
**Pull request by sptramer**
## Change summary

Instructions on how to create a binary build that you can distribute so that two teams (project and engine) can have separated workflows.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by amzn-pratikpa — APPROVED** [review]

File: content/docs/user-guide/build/distributable-engine.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+1. Install the version of Python and modules required by the engine.
+   
+   ```cmd
+   python\get_python.bat
+   ```
+
+### Register the engine
+
+The next step in creating a distributable build of the engine tools is to register the locally installed engine. These steps should be re-run every time you create a new build of the `INSTALL` target.
```
Comment (chanmosq) [inline content/docs/user-guide/build/distributable-engine.md:86]:
```suggestion
The next step in creating a distributable build of the engine tools is to register the locally installed engine. You must re-run these steps every time you create a new build of the `INSTALL` target.
```

**Review by chanmosq — APPROVED** [review]

**Comment by chanmosq** [comment]
Approved with a few suggestions.

File: content/docs/user-guide/build/distributable-engine.md (hunk 1 above)
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:86]: I'm not sure why you'd need to re-register the engine every time you build it to the install target, assuming the engine name and install path are kept the same. I'm curious as to the reason for stating this requirement.

File: content/docs/user-guide/build/distributable-engine.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+1. Register the project with the _local development engine_. Run the following command from the directory containing the engine source:
+   
+   ```cmd
+   scripts\o3de.bat register -pp W:\MyProject
+   ```
+1. Follow the [project build instructions](https://o3de.org/docs/user-guide/build/) to generate any project-specific builds, such as the launcher.
+
+### Project developer setup
+
+Project developers will work on project-specific source code or use O3DE creative tools to add project content. To enable their workflows, project developers are required to _only_ pull a project from its hosted location.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:224]: I suggest rewording this sentence. I'm not 100% sure what is really meant here. The emphasis of the word _only_, together with its location in the sentence, is awkward. It seems to be trying to convey that project developers _must_ pull a project from its hosted location, and no other location. If this is truly the case, then I'd start by replacing "project developers are required to only" with "project developers must only". Then, at the beginning of that sentence, in place of "To enable their workflows", I'd supply a more precise reason why they must do this.

**Review by willihay — APPROVED** [review]
Approving in order to expedite, but I recommend a few clarifications be made now or in a fast follow.

File: content/docs/user-guide/build/distributable-engine.md (hunk 1 above)
Comment (lumberyard-employee-dm) [reply content/docs/user-guide/build/distributable-engine.md:86]: The engine only need to be registered if the `LY_VERSION_ENGINE_NAME` value was changed as that effects the engine name in the install layout

File: content/docs/user-guide/build/distributable-engine.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    ```
+
+    This creates the binaries in an `install` subdirectory of your project. For our example it would be `W:\MyProject\install`.
+
+## Distribute your build
+
+At this point, the created `install` directory is your final set of distributables. It's up to your team to decide where to host the build, but we recommend a source control system that has good blob support.
+
+{{< important >}}
+The .gitignore rule automatically adds the following folders to ignore for git. If using another source control system, make sure the following subdirectories of your project are ignored and not checked in:
```
Comment (lumberyard-employee-dm) [inline content/docs/user-guide/build/distributable-engine.md:186]: It may be good to notate that the .gitignore file is located in the project root after project creation and can be referenced to determine the list of directories/files  to ignore

File: content/docs/user-guide/build/distributable-engine.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+1. Install the version of Python and modules required by the engine.
+   
+   ```cmd
+   python\get_python.bat
+   ```
+
+### Register the engine
+
+The next step in creating a distributable build of the engine tools is to register the locally installed engine. You must re-run these steps every time you create a new build of the `INSTALL` target.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:86]:
```suggestion
The next step in creating a distributable build of the engine tools is to register the locally installed engine. You must re-run this registration step any time you make a change to the CMake cache variable `LY_VERSION_ENGINE_NAME` or to the `engine_name` in `engine.json`.
```

Comment (willihay) [reply content/docs/user-guide/build/distributable-engine.md:86]: @lumberyard-employee-dm How's this for a rewording?

File: content/docs/user-guide/build/distributable-engine.md (hunk 2 above)
Comment (lumberyard-employee-dm) [reply content/docs/user-guide/build/distributable-engine.md:224]: I hesitate calling this "Project developer setup".

This seems like information that is in addition to the engine of the "Engine development setup" section.
For users that don't need to re-build the project, they can use the prebuild binaries in the `<project-root>/install/bin/Windows/profile/Default` subdirectory.

File: content/docs/user-guide/build/distributable-engine.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    ```
+
+    This creates the binaries in an `install` subdirectory of your project. For our example it would be `W:\MyProject\install`.
+
+## Distribute your build
+
+At this point, the created `install` directory is your final set of distributables. It's up to your team to decide where to host the build, but we recommend a source control system that has good blob support.
+
+{{< important >}}
+The .gitignore rule automatically adds the following folders to ignore for git. If using another source control system, make sure the following subdirectories of your project are ignored and not checked in:
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:186]:
```suggestion
The .gitignore rule automatically adds the following folders to ignore for git. The `.gitignore` file is located in the project root after project creation. If using another source control system, make sure the following subdirectories of your project are ignored and not checked in:
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+    
+    ```
+    \user
+    \build
+    \Cache
+    ```
+
+These files are generated locally during builds, and checking them in may cause issues for developers.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:194]:
```suggestion
These files are generated locally during builds, and checking them in may cause issues for developers. Reference the `.gitignore` file in the project root to verify the list of directories and files to ignore.
```

**Review by willihay — APPROVED** [review]

File: content/docs/user-guide/build/distributable-engine.md (hunk 4 above)
Comment (lumberyard-employee-dm) [reply content/docs/user-guide/build/distributable-engine.md:86]: Could we clarify that the engine.json is located at `<engine-root>/engine.json` or something that points the user to the engine source root directory?
Users might not associate the `engine.json` file with being inside of the engine root immediately

Besides that, the rewording of that section is good.

Comment (willihay) [reply content/docs/user-guide/build/distributable-engine.md:86]: Actually, to avoid the potential for confusion here between the source `engine.json` and the installed `engine.json`, I'm going to remove the mention of `engine.json` in my rewording, especially since these instructions have been using `LY_VERSION_ENGINE_NAME`.

File: content/docs/user-guide/build/distributable-engine.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+At this point, the installed engine build is now registered in the O3DE config directory (by default `%USERPROFILE%\.o3de`) in the `o3de_manifest.json` file.
+
+## Project association
+
+At this point, you're ready to either create a new project for distribution or re-associate an existing project with the install build that will be used to create the distributable.
+
+### Create a new project
+
+Run the following command from the local engine source directory:
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:106]: Wait, shouldn't the new project be created from the distributable build directory, not the engine source directory?

Comment (lumberyard-employee-dm) [reply content/docs/user-guide/build/distributable-engine.md:106]: It should be created from the distributable install directory.

Using the engine source directory would result in the same project being created, BUT with the project being registered using the source engine name and not the SDK layout engine name which was set via the `LY_ENGINE_VERSION_NAME`

Comment (willihay) [reply content/docs/user-guide/build/distributable-engine.md:106]: Yes! That's exactly what I was worried about. Thanks for verifying. Will update.

**Review by willihay — APPROVED** [review]

File: content/docs/user-guide/build/distributable-engine.md (hunk 2 above)
Comment (willihay) [reply content/docs/user-guide/build/distributable-engine.md:224]: Leaving this for @sptramer to address in a follow up.
